BraintreeCore/src/main/java/com/braintreepayments/api/core/PaymentMethod.kt

```diff
@@ -1,103 +1,38 @@
 package com.braintreepayments.api.core
 
-import android.os.Parcel
 import androidx.annotation.RestrictTo
 import org.json.JSONException
 import org.json.JSONObject
 
-/**
- * An abstract class to extend when creating a payment method. Contains logic and
- * implementations shared by all payment methods.
- */
-abstract class PaymentMethod {
+interface PaymentMethod {
 
-    /**
-     * @suppress
-     */
-    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
-    companion object {
+    var sessionId: String?
+    var source: String?
+    var integration: String?
+    val apiPath: String
 
+    companion object {
         const val OPERATION_NAME_KEY = "operationName"
         const val OPTIONS_KEY = "options"
         const val VALIDATE_KEY = "validate"
-
-        private const val DEFAULT_SOURCE = "form"
-        private const val DEFAULT_INTEGRATION = "custom"
-    }
-
-    private var _sessionId: String? = null
-    private var _source: String? = DEFAULT_SOURCE
-    private var _integration: String? = DEFAULT_INTEGRATION
-
-    abstract val apiPath: String?
-
-    /**
-     * @suppress
-     */
-    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
-    constructor()
-
-    /**
-     * Sets the integration method associated with the tokenization call for analytics use.
-     * Defaults to custom and does not need to ever be set.
-     *
-     * @param integration the current integration style.
-     * @suppress
-     */
-    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
-    fun setIntegration(integration: String) {
-        _integration = integration
-    }
-
-    /**
-     * Sets the source associated with the tokenization call for analytics use. Set automatically.
-     *
-     * @param source the source of the payment method.
-     * @suppress
-     */
-    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
-    fun setSource(source: String) {
-        _source = source
-    }
-
-    /**
-     * @param sessionId sets the session id associated with this request. The session is a uuid.
-     * This field is automatically set at the point of tokenization, and any previous
-     * values ignored.
-     * @suppress
-     */
-    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
-    fun setSessionId(sessionId: String?) {
-        // TODO: require session id in constructor to eliminate the possibility of a null value
-        _sessionId = sessionId
+        const val DEFAULT_SOURCE = "form"
+        const val DEFAULT_INTEGRATION = "custom"
     }
 
     @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
     fun buildMetadataJSON(): JSONObject {
         return MetadataBuilder()
-            .sessionId(_sessionId)
-            .source(_source)
-            .integration(_integration)
+            .sessionId(sessionId)
+            .source(source)
+            .integration(integration)
             .build()
     }
```

Comment:
It might be better to duplicate this logic in each class that implements `PaymentMethod` for now. I'm not sure if Java fully supports default implementation [just yet](https://stackoverflow.com/a/49933841). Even if there is support for it, I think ultimately we'll want PaymentMethod types to become Plain Data objects.